feat: Auto-activate Scene 6 (Midnight Mode) when brightness is 1-3%

jokerigno opened the pull request:
This adds automatic Scene 6 activation for Philips ZhiRui lamps when brightness is set to very low levels (1-3%), providing better UX for activating midnight mode via UI sliders.

- Detects brightness 1-3% and auto-switches to Scene 6
- Includes error handling and fallback
- No breaking changes
- Tested on philips.light.moonlight

jokerigno committed:
feat: Auto-activate Scene 6 (Midnight Mode) when brightness is 1-3%

This adds automatic Scene 6 activation for Philips ZhiRui lamps when
brightness is set to very low levels (1-3%), providing better UX for
activating midnight mode via UI sliders.

- Detects brightness 1-3% and auto-switches to Scene 6
- Includes error handling and fallback
- No breaking changes
- Tested on philips.light.moonlight

syssi commented:
Thanks for your contribution! I will review your merge request as soon as possible.

syssi commented on custom_components/xiaomi_miio_philipslight/light.py:
```diff
@@ -927,6 +927,30 @@ async def async_turn_on(self, **kwargs):
             brightness = kwargs[ATTR_BRIGHTNESS]
             percent_brightness = ceil(100 * brightness / 255.0)
 
+            # === Auto-activate Scene 6 (Midnight Mode) when brightness is 1-3% ===
```
```suggestion
```

syssi commented on custom_components/xiaomi_miio_philipslight/light.py:
```diff
@@ -927,6 +927,30 @@ async def async_turn_on(self, **kwargs):
+                    self._name
+                )
+
+                # Activate scene 6
```
```suggestion
```

syssi commented on custom_components/xiaomi_miio_philipslight/light.py:
```diff
@@ -927,6 +927,30 @@ async def async_turn_on(self, **kwargs):
+                    self._state = True
+                    self._brightness = brightness
+                    self._scene = 6
+                    return  # Exit without calling set_brightness
```
```suggestion
                    return
```

syssi commented on custom_components/xiaomi_miio_philipslight/light.py:
```diff
@@ -927,6 +927,30 @@ async def async_turn_on(self, **kwargs):
+                    return  # Exit without calling set_brightness
+                else:
+                    _LOGGER.error("Failed to set scene 6 for %s, falling back to brightness", self._name)
+            # === End of custom modification ===
```
```suggestion
```

syssi commented on custom_components/xiaomi_miio_philipslight/light.py:
```diff
@@ -927,6 +927,30 @@ async def async_turn_on(self, **kwargs):
             percent_brightness = ceil(100 * brightness / 255.0)
 
+            # === Auto-activate Scene 6 (Midnight Mode) when brightness is 1-3% ===
+            if percent_brightness >= 1 and percent_brightness <= 3:
```
Could you add a configurable opt-out option if somebody doesn't like this improvement?

jokerigno committed:
Add configurable opt-out for auto-midnight mode feature

Added `auto_midnight_mode` configuration option (default: True) to allow
users to disable the automatic Scene 6 activation when brightness is set
to 1-3%.

Users can opt-out by adding to their configuration:
```yaml
auto_midnight_mode: false
```

This addresses the maintainer's request for a configurable opt-out option.

jokerigno commented on custom_components/xiaomi_miio_philipslight/light.py:
```diff
@@ -171,7 +174,7 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     elif model == "philips.light.moonlight":
         light = PhilipsMoonlight(host, token)
-        device = XiaomiPhilipsMoonlightLamp(name, light, model, unique_id)
+        device = XiaomiPhilipsMoonlightLamp(name, light, model, unique_id, auto_midnight_mode)
```
Feature is active by default to disable add false:

- platform: xiaomi_miio_philipslight
  host: ...
  token: ...
  auto_midnight_mode: false  # disable  scene 6 when brightness is 1-3%

jokerigno committed:
Update custom_components/xiaomi_miio_philipslight/light.py

Co-authored-by: Sebastian Muszynski <basti@linkt.de>

jokerigno committed:
Update custom_components/xiaomi_miio_philipslight/light.py

Co-authored-by: Sebastian Muszynski <basti@linkt.de>

jokerigno committed:
Update custom_components/xiaomi_miio_philipslight/light.py

Co-authored-by: Sebastian Muszynski <basti@linkt.de>

jokerigno committed:
Update custom_components/xiaomi_miio_philipslight/light.py

Co-authored-by: Sebastian Muszynski <basti@linkt.de>

syssi commented on custom_components/xiaomi_miio_philipslight/light.py:
```diff
@@ -131,6 +133,7 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     token = config[CONF_TOKEN]
     name = config[CONF_NAME]
     model = config.get(CONF_MODEL)
+    auto_midnight_mode = config.get(CONF_AUTO_MIDNIGHT_MODE, True)
```
```suggestion
    auto_midnight_mode = config.get(CONF_AUTO_MIDNIGHT_MODE)
```

syssi committed:
Update custom_components/xiaomi_miio_philipslight/light.py

syssi commented on custom_components/xiaomi_miio_philipslight/light.py:
```diff
@@ -798,12 +801,13 @@ class XiaomiPhilipsMoonlightLamp(XiaomiPhilipsBulb):
     _attr_supported_color_modes = {ColorMode.COLOR_TEMP, ColorMode.HS}
 
-    def __init__(self, name, light, model, unique_id):
+    def __init__(self, name, light, model, unique_id, auto_midnight_mode=True):
```
```suggestion
    def __init__(self, name, light, model, unique_id, auto_midnight_mode):
```

syssi committed:
Update custom_components/xiaomi_miio_philipslight/light.py

syssi committed:
Fix lint issues

syssi reviewed (approved):
This is good to go! :-)
